### PR TITLE
Gradient Accumulation

### PR DESCRIPTION
**Pull request by hktxt**
Hi~ 
I found that SMP does not support gradient acumulation, which is quite useful for those who has limited compute resources. So I made a modification to support it.

### CODE REVIEW

**Comment by howard-mahe** [comment]
Hello, I don't think this library contains instructions to raise warnings until now.
However, considering that (1) gradient accumulations can lead to terrible performance when the batch size is small and (2) fine-tuning the Batch Normalization layers of the backbone is the default behavior in this library, I think this feature should raise a warning which would encourage the user to disable Batch Normalization fine-tuning when he uses gradient accumulation.
